miner/worker.go

```diff
@@ -360,13 +368,45 @@ func (miner *Miner) applyTransaction(env *environment, tx *types.Transaction) (*
 		gp   = env.gasPool.Gas()
 	)
 	receipt, err := core.ApplyTransaction(miner.chainConfig, miner.chain, &env.coinbase, env.gasPool, env.state, env.header, tx, &env.header.GasUsed, vm.Config{})
+	// If successful, and not just reproducing the block, check the interop executing messages.
+	if err == nil && !env.noTxs && miner.chain.Config().IsInterop(env.header.Time) {
+		// Whenever there are `noTxs` it means we are building a block from pre-determined txs. There are two cases:
+		//	(1) it's derived from L1, and will be verified asynchronously by the op-node.
+		//	(2) it is a deposits-only empty-block by the sequencer, in which case there are no interop-txs to verify (as deposits do not emit any).
+		err = miner.checkInterop(env.rpcCtx, tx, receipt)
+	}
 	if err != nil {
 		env.state.RevertToSnapshot(snap)
 		env.gasPool.SetGas(gp)
 	}
 	return receipt, err
 }
 
+func (miner *Miner) checkInterop(ctx context.Context, tx *types.Transaction, receipt *types.Receipt) error {
+	b, ok := miner.backend.(BackendWithInterop)
+	if !ok {
+		return fmt.Errorf("cannot mine interop txs without interop backend, got backend type %T", miner.backend)
+	}
+	if ctx == nil { // check if the miner was set up correctly to interact with an RPC
+		return errors.New("need RPC context to check executing messages")
+	}
+	executingMessages, err := interoptypes.ExecutingMessagesFromLogs(receipt.Logs)
+	if err != nil {
+		return fmt.Errorf("cannot parse interop messages from receipt of %s: %w", receipt.TxHash, err)
+	}
+	if len(executingMessages) == 0 {
+		return nil // avoid an RPC check if there are no executing messages to verify.
+	}
+	if err := b.CheckMessages(ctx, executingMessages, interoptypes.CrossUnsafe); err != nil {
+		if ctx.Err() != nil { // don't reject transactions permanently on RPC timeouts etc.
+			return err
+		}
+		tx.SetRejected() // Mark the tx as rejected: it will not be welcome in the tx-pool anymore.
```

Comment:
We will want to monitor how often this happens, don't need that functionality in this PR necessarily

Reply:
added a metric